File: With_Levels/Bronze/Bronze_1/5195.cpp
// // problem: Stump Speech
// // id: 5195
// // tag:
// // time taken:
// #include <bits/stdc++.h>
// using namespace std;
// int main(void)
// {
//     ios::sync_with_stdio(false);
//     cin.tie(nullptr);

//     int tc, n, p, ans, l, wl;
//     string s, t;
//     bool chk;
//     cin >> tc >> n;
//     for (int tcc = 1; tcc <= tc; tcc++)
//     {
//         map<string, int> m;
//         ans = 0;
//         for (int i = 0; i < n; i++)
//         {
//             cin.ignore();
//             getline(cin, s);
//             cin >> p;
//             m[s] = p;
//         }
//         cin.ignore();
//         string t = "";
//         while (getline(cin, s))
//         {
//             if (s.length() == 1 && isdigit(s[0]))
//                 break;
//             t += ' ';
//             t += s;
//         }

//         l = t.length();
//         for (auto w : m)
//         {
//             wl = w.first.length();
//             for (int i = 0; i < l - wl + 1; i++)
//             {
//                 if (t.substr(i, wl) == w.first)
//                     ans += w.second;
//             }
//         }
//         cout << "Data Set " << tcc << ":\n"
//              << ans << '\n';
//     }

//     return 0;
// }

File: With_Levels/Bronze/Bronze_1/5195.py
# problem: Stump Speech
# id: 5195
# tag:
# time taken:

for cnt in range(1, 1 + int(input())):
    dic = {}
    n = int(input())
    for _ in range(n):
        word = input()
        dic[word] = int(input())

    ans = 0
    msg = input()
    l = len(msg)
    for k, v in dic.items():
        wl = len(k)
        for i in range(l - wl + 1):
            match = True
            for j in range(wl):
                if k[j] != msg[i + j]:
                    match = False
                    break
            if match:
                ans += v
    print(f"Data Set {cnt}:")
    print(ans)
